src/kernel/scheduling/task.c

```diff
@@ -7,9 +8,56 @@
 task_t* create_task(char *name, void (*_entry_point)(void *), void *args) {
     task_t* new_task = (task_t*) kmalloc(sizeof(task_t));
     strcpy(new_task->task_name, name);
-    loglinef(Verbose, "Task created with name: %s", new_task->task_name);
-    thread_t* thread = create_thread(name, _entry_point, args);
-    new_task->threads = thread;
     new_task->parent = NULL;
+    new_task->task_id = next_task_id++;
+    loglinef(Verbose, "Task created with name: %s - Task id: %d", new_task->task_name, new_task->task_id);
+    thread_t* thread = create_thread(name, _entry_point, args, new_task);
+    new_task->threads = thread;
+    scheduler_add_task(new_task);
     return new_task;
 }
+
+bool add_thread_to_task_by_id(size_t task_id, thread_t* thread) {
+    task_t* task = get_task(task_id);
+    if (task_id > next_task_id || task == NULL) {
```

Comment:
This if fine, but I do have a suggestion:
`get_task()` returns NULL if the task wasnt found, maybe check for that instead of checking if the task id > the next task id. 
My thinking here is that if you ever decide to recycle task ids, you only need to update get_task to check for that, and not any places.